bindings/pyroot/pythonizations/test/rdataframe_asnumpy.py

```diff
@@ -280,10 +280,34 @@ def test_pickle(self):
         npy = df.AsNumpy(["x"])
         arr = npy["x"]
 
-        pickle.dump(arr, open("rdataframe_asnumpy.pickle", "wb"))
-        arr2 = pickle.load(open("rdataframe_asnumpy.pickle", "rb"))
+        with open("rdataframe_asnumpy.pickle", "wb") as f:
+            pickle.dump(arr, f)
+        with open("rdataframe_asnumpy.pickle", "rb") as f:
+            arr2 = pickle.load(f)
         self.assertTrue(all(arr == arr2))
 
+    def test_memory_adoption_fundamental_types(self):
+        """
+        Testing the adoption of the memory from the C++ side for fundamental types
+        """
+        df = ROOT.ROOT.RDataFrame(1).Define("x", "1.0")
+        npy = df.AsNumpy(["x"])
+        pyarr = npy["x"]
+        cpparr = pyarr.result_ptr.GetValue()
```

Comment:
This `result_ptr` is RDataFrame's result pointer that you store?

Reply:
yep, exactly!